src/pages/ReportsUpload/index.tsx

```diff
@@ -0,0 +1,341 @@
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import {
+    Button,
+    Modal,
+    ButtonVariant,
+    Bullseye,
+    EmptyState,
+    EmptyStateVariant,
+    EmptyStateIcon,
+    Title,
+    EmptyStateBody,
+    ProgressMeasureLocation,
+    Progress,
+    TitleLevel,
+    ProgressVariant,
+    EmptyStateSecondaryActions
+} from '@patternfly/react-core';
+import { VolumeIcon } from '@patternfly/react-icons';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import * as uploadActions from '../../actions/UploadActions';
+import { GlobalState } from '../../models/state';
+import { RouterGlobalProps } from '../../models/router';
+import {
+    Main,
+    PageHeader,
+    PageHeaderTitle
+} from '@redhat-cloud-services/frontend-components';
+import { Upload } from '../../models';
+import './ReportsUpload.scss';
+import { UploadForm } from './UploadForm';
+
+interface FormValues {
+    file: string;
+    reportName: string;
+    reportDescription: string;
+    yearOverYearGrowthRatePercentage: number;
+    percentageOfHypervisorsMigratedOnYear1: number;
+    percentageOfHypervisorsMigratedOnYear2: number;
+    percentageOfHypervisorsMigratedOnYear3: number;
+    percentageOfHypervisorsMigratedSum: number // Sum of year1, year2, and year3
+}
+
+interface StateToProps {
+    file: File;
+    success: boolean | null;
+    error: string | null;
+    progress: number;
+    uploading: boolean
+}
+
+interface DispatchToProps {
+    uploadProgress: (progress: number) => void;
+    uploadRequest: (upload: Upload, config: {}) => void;
+    selectUploadFile: (file: File) => void;
+}
+
+interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
+}
+
+interface State {
+    showForm: boolean;
+    timeoutToRedirect: number;
+}
+
+export class ReportsUpload extends React.Component<Props, State> {
+
+    redirectTimer: any;
+    beforeUnloadHandler: any;
+
+    initialFormValue: FormValues;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            showForm: true,
+            timeoutToRedirect: 3
+        };
+
+        this.initialFormValue = {
```

Comment:
It looks like `initialFormValue` doesn't need to be a class property, it's a little strange to put it there I think. You could just define it as a `const` just above your class, it could still use the `FormValues` type.

Reply:
It makes sense to have `initialFormvalue` outside the constructor. Thanks!